dataland-frontend/tests/e2e/utils/EuTaxonomyNonFinancialsUpload.ts

```diff
@@ -13,23 +13,68 @@ import Chainable = Cypress.Chainable;
  * Uploads a single eutaxonomy-non-financials data entry for a company via the Dataland upload form
  *
  * @param companyId The Id of the company to upload the dataset for
+ * @param valueFieldNotFilled Value which, if true, disables the value field
  * @returns the id of the dataset that has been uploaded
  */
-export function uploadEuTaxonomyDataForNonFinancialsViaForm(companyId: string): Cypress.Chainable<string> {
+export function uploadEuTaxonomyDataForNonFinancialsViaForm(
+  companyId: string,
+  valueFieldNotFilled = false
+): Cypress.Chainable<string> {
   cy.visitAndCheckAppMount(`/companies/${companyId}/frameworks/${DataTypeEnum.EutaxonomyNonFinancials}/upload`);
-  cy.get(`input[name="reportingPeriod"]`).type("2023");
-  cy.get("select[name=assurance]").select("Limited Assurance");
-  cy.get('input[id="reportingObligation-option-yes"][value=Yes]').check({
-    force: true,
+
+  cy.get('button[data-test="upload-files-button"]').click();
+  cy.get("input[type=file]").selectFile("tests/e2e/fixtures/pdfTest.pdf", { force: true });
+  cy.get('div[data-test="uploaded-files"]').find('[data-test="uploaded-files-title"]').should("contain", "pdf");
+  cy.get('div[data-test="uploaded-files"]').find('[data-test="uploaded-files-size"]').should("contain", "KB");
+  cy.get('button[data-test="uploaded-files-remove"]').click();
+  cy.get('div[data-test="uploaded-files"]').should("not.exist");
+
+  cy.get('button[data-test="upload-files-button"]').click();
+  cy.get("input[type=file]").selectFile("tests/e2e/fixtures/pdfTest.pdf", { force: true });
+  cy.get('div[data-test="uploaded-files"]')
+    .should("exist")
+    .find('[data-test="uploaded-files-title"]')
+    .should("contain", "pdf");
+  cy.get('input[name="currency"]').type("qqq");
+
+  cy.get('[data-test="fiscalYearEnd"] button').should("have.class", "p-datepicker-trigger").click();
+  cy.get("div.p-datepicker").find('button[aria-label="Next Month"]').click();
+  cy.get("div.p-datepicker").find('span:contains("11")').click();
+  cy.get('input[name="fiscalYearEnd"]').invoke("val").should("contain", "11");
+  cy.get('input[name="fiscalYearDeviation"][value="Deviation"]').check();
+  cy.get('div[id="jumpLinks"] li:last a').click();
+  cy.window().then((win) => {
+    const scrollPosition = win.scrollY;
+    expect(scrollPosition).to.be.greaterThan(0);
   });
-  for (const argument of ["capex", "opex"]) {
-    cy.get(`div[title=${argument}] input`).each(($element, index) => {
-      const inputNumber = 10 * index + 7;
-      cy.wrap($element).type(inputNumber.toString(), { force: true });
+  cy.get('[data-test="fiscalYearEnd"] button').should("have.class", "p-datepicker-trigger").should("exist");
+  cy.get('input[name="fiscalYearEnd"]').should("not.be.visible");
```

Comment:
Same comment as for the EuTaxoFinancialsUpload.ts =>

This is pure UI testing which could happen in a component test.

The actual filling of the form can and should stay here.  Only then the function name is also ok, because currently it does more than just filling the form.

Reply:
We will investigate/do this.